app/src/main/kotlin/com/machinerychorus/lifeprogresswallpaper/WallpaperSettingsFragment.kt

```diff
@@ -15,6 +21,50 @@ class WallpaperSettingsFragment : PreferenceFragmentCompat() {
 		//we set it here to work around that
 		val textPref = findPreference<EditTextPreference>(getString(R.string.goalsKey))
 		textPref?.setOnBindEditTextListener { editText -> editText.setHint(R.string.goalsHint) }
+
+		val manager = ColorPickerPreferenceManager.getInstance(context)
+
+		// listeners to keep hex and color selector in sync
+		val bgPref = findPreference<ColorPreference>(getString(R.string.bgColorKey))
+		val bgHexPref = findPreference<EditTextPreference>(getString(R.string.bgColorHexKey))
+		bgPref?.getColorPickerView()?.setColorListener(ColorEnvelopeListener { envelope, _ ->
+			bgHexPref?.text = "#"+envelope.hexCode
+		})
+		bgHexPref?.setOnPreferenceChangeListener { _, newValue ->
+			manager.clearSavedColor(getString(R.string.bgColorKey))
+			manager.clearSavedSelectorPosition(getString(R.string.bgColorKey))
+			manager.clearSavedAlphaSliderPosition(getString(R.string.bgColorKey))
+			manager.clearSavedBrightnessSlider(getString(R.string.bgColorKey))
+			manager.setColor(getString(R.string.bgColorKey), Color.parseColor(newValue.toString()))
```

Comment:
You need to add a try+catch around `Color.parseColor`, or else the app will crash when the input can't be parsed as a color. Stacktrace: 
<details>

```
java.lang.IllegalArgumentException: Unknown color
    at android.graphics.Color.parseColor(Color.java:1391)
    at com.machinerychorus.lifeprogresswallpaper.WallpaperSettingsFragment.onCreatePreferences$lambda$3(WallpaperSettingsFragment.kt:38)
    at com.machinerychorus.lifeprogresswallpaper.WallpaperSettingsFragment.$r8$lambda$Unk1E1zNiavyZJLz4QMWb0D3Y80(Unknown Source:0)
    at com.machinerychorus.lifeprogresswallpaper.WallpaperSettingsFragment$$ExternalSyntheticLambda2.onPreferenceChange(Unknown Source:6)
    at androidx.preference.Preference.callChangeListener(Preference.java:1131)
    at androidx.preference.EditTextPreferenceDialogFragmentCompat.onDialogClosed(EditTextPreferenceDialogFragmentCompat.java:149)
    at androidx.preference.PreferenceDialogFragmentCompat.onDismiss(PreferenceDialogFragmentCompat.java:291)
    at androidx.fragment.app.DialogFragment$3.onDismiss(DialogFragment.java:132)
    at android.app.Dialog$ListenersHandler.handleMessage(Dialog.java:1393)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:193)
    at android.app.ActivityThread.main(ActivityThread.java:6669)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

</details>

The minimum here IMO is to put a try+catch around the parse call, and pop up a toast to alert the user if there's an error. Check out [IntegerPreference,setText](https://github.com/ethanmdavidson/DeathProgress/blob/main/app/src/main/kotlin/com/machinerychorus/lifeprogresswallpaper/customPrefs/IntegerPreference.kt#L21) for an example.